File: mobile/lib/features/categories/domain/entities/category_entity.dart
import 'package:equatable/equatable.dart';

class CategoryEntity extends Equatable {
  final String id;
  final String userId;
  final String name;
  final String type; // INCOME, EXPENSE, TRANSFER
  final String? icon;
  final String? color;
  final bool isDefault;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  const CategoryEntity({
    required this.id,
    required this.userId,
    required this.name,
    required this.type,
    this.icon,
    this.color,
    required this.isDefault,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        name,
        type,
        icon,
        color,
        isDefault,
        isActive,
        createdAt,
        updatedAt,
      ];
}

File: backend/src/modules/categories/entities/category.entity.ts
export class CategoryEntity {
  id!: string;
  userId!: string;
  name!: string;
  type!: string; // INCOME, EXPENSE, TRANSFER
  icon?: string | null;
  color?: string | null;
  isDefault!: boolean;
  isActive!: boolean;
  createdAt!: Date;
  updatedAt!: Date;
  deletedAt?: Date | null;

  constructor(partial: Partial<CategoryEntity>) {
    Object.assign(this, partial);
  }
}
